### DISCUSSION TITLE
Wifi -> RPi Zero W -> USB tethering -> Tesla-Android

### DISCUSSION THREAD

**Discussion by RetoHB** [tesla-android/issue-tracker#265] [Q&A]
I had a new idea on how to get wireless Internet to Tesla Android without using an LTE dongle.
I have a PRi Zero W which connects to my Wifi Internet. On the Zero I have gnirehtet running https://github.com/Genymobile/gnirehtet
Tesla Android then connects to the Zero via USB tethering. 

Wifi -> RPi Zero W -> USB tethering -> Tesla-Android (-> Wifi Tesla)

In principle, it seems to work. However, gnirehtet uses a VPN connection. As soon as I allow the VPN connection, the Wifi connection from Tesla-Android to the car breaks. I assume that the VPN connection is tapping everything, including the Tesla-Android's Wifi and thus the connection is lost.

Is there a way to limit the VPN part to USB tethering only and leave the Tesla-Android Wifi alone?

**Comment by RetoHB** [comment]
Finally I found a solution that works for me! I replaced the USB connection with an Ethernet hat and an RJ45 cable!

So the PRi Zero W connects to my Wifi Hotspot and shares the Internet connection to the Tesla-Android RPi4.
Works like a charm. Especially now with the 5GHz connection to the Tesla! All smooth!

![image](https://github.com/tesla-android/issue-tracker/assets/9882700/b843b1bc-3d30-448e-9a17-85bcf020fe2d)

**Reply by Michxik** [reply]
So u use internet from your phone and pi zero w connects to it and then it goes to android tesla, and you can use internet on tesla android? like watching stuff online?

**Reply by RetoHB** [reply]
I am not sure about the Internet speed in the Tesla. But I do get traffic data can use the browser and listen to Spotify. For video streaming I would disconnect Tesla Android and use the Tesla Internet connection anyway.

**Reply by Michxik** [reply]
the thing is that I would like to watch some movies as a passenger while the car is riding. Is there any way to use internet from tesla lte and not from for example my phone?

**Reply by RetoHB** [reply]
No

**Reply by Michxik** [reply]
So can I make hotspot from my phone and connect android auto to it? without this raspberry pi zero w? or do I need your solution?

**Reply by RetoHB** [reply]
Having two Wifi connections at the same time, one to the car and one to your phone is not possible. Bersides mine I am not aware of any other solution for wireless internet access.

**Reply by GtHbbrr** [reply]
I can use the hot spot from my phone (iphone) over USB to the RPI4 that Tesla Android is running on. No need for the extra RPIZero W. Using the hot spot over USB has some specific quirks to get this to work. It seems that for iphone the hot-spot-over-USB only works when I use an original Apple lightning to USB cable and when I first connect the iphone to the RPI4 USB before connecting power to the RPI4 to start Tesla Android

**Reply by RetoHB** [reply]
Sure this works. But I do not want to connect the cable to my phone every time I enter the car.

**Comment by NKPenfold** [comment]
Hi RetoHB - this looks like just the solution I need!! And I have a Pi Zero kicking about in a drawer in my office. Before I go ahead and start trying to get it working for me can I ask what OS you run on the Pi Zero that you then have gnirehtet running on? Or does it not matter too much?

Great solution by the way. At the moment I have my phone usb tethered to the Pi to give Tesla Android internet access which is a pain to keep plugging in an unplugging

**Reply by RetoHB** [reply]
Really not sure what I used back then. But probably just the image Pi imager offers...

**Reply by NKPenfold** [reply]
Apologies for another question but I thought I'd ask while I wait for my pi zero and ethernet hat to arrive from Amazon. How did you power the pi zero? Did you use a separate usb cable or did you use the gpio?

**Reply by RetoHB** [reply]
Check the red/white cable in my picture.
